src/docs/arch-cursor.dox

```diff
@@ -3,13 +3,393 @@
 Cursors are used in WiredTiger to get and modify data.
 A caller of WiredTiger uses WT_SESSION::open_cursor to create
 a WT_CURSOR.  Methods on the WT_CURSOR can then be used to
-position, iterate, get, and set data.
+position, iterate, get, and set data.  In the typical case, a cursor
+will be used to access keys and values in a Btree.  Cursors are
```

Comment:
However, cursors can also be used to access indexed data, WiredTiger statistics, log files,
and metadata.